tuxemon/event/actions/set_code.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import final
+
+from tuxemon.event.eventaction import EventAction
+from tuxemon.locale import T
+from tuxemon.menu.input import InputMenu
+
+
+@final
+@dataclass
+class SetCodeAction(EventAction):
+    """
+    Set a code and checks if it's correct or not.
+    Case Sensitive:
+    ATTENTION and AtTenTION are two different words.
+
+    Script usage:
+        .. code-block::
+
+            set_code <code>,<variable>
+
+    Script parameters:
+        code: The actual code the player needs to know.
+        variable: Where the result (on/off) is saved.
+
+    """
+
+    name = "set_code"
+    code: str
+    variable: str
+
+    def check_setcode(self, name: str) -> None:
+        player = self.session.player
+        if self.code == name:
+            player.game_variables[self.variable] = "on"
+        else:
+            player.game_variables[self.variable] = "off"
+
+    def start(self) -> None:
+        self.session.client.push_state(
+            state_name=InputMenu,
+            prompt=T.translate("input_code"),
+            callback=self.check_setcode,
```

Comment:
We should probable make the prompt a parameter. I.e. don't hardcode, "input_code" here.

Reply:
No problem